### PR TITLE
Program security - PDA sharing lesson updated

### PR DESCRIPTION
**Pull request by 0xCipherCoder**
### Summary of Changes

- Updated code snippets with the latest anchor version
- Fixed content with updated version changes
- Fixed grammar and styling
- Fixed as per guidelines

Fixes #

Unboxed PRs
Starter - https://github.com/solana-developers/pda-sharing/pull/1
Solution - https://github.com/solana-developers/pda-sharing/pull/2

### CODE REVIEW

File: content/courses/program-security/pda-sharing.md [hunk 1]
```diff
@@ -189,155 +196,168 @@ pub mod pda_sharing_recommended {
...
+      .accounts({
+        pool: insecurePoolFake.publicKey,
+        mint: tokenMint,
+        vault: insecureVault.address,
+        withdrawDestination: fakeWithdrawDestination,
+      })
+      .signers([insecurePoolFake])
+      .rpc();
+
+    await spl.mintTo(
```
Comment (mikemaccana) [inline content/courses/program-security/pda-sharing.md:292]: Just import `mintTo`, to match the other lessons.

Comment (mikemaccana) [reply content/courses/program-security/pda-sharing.md:292]: PS nice work getting rid of the unnecessary delay in the old code.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Great work, some minor changes but not many, we should be able to get this in immediately once done.

File: content/courses/program-security/pda-sharing.md [hunk 2]
```diff
@@ -63,28 +63,31 @@ pub struct WithdrawTokens<'info> {
...
-        };
-        CpiContext::new(program, accounts)
+        CpiContext::new(
+            self.token_program.to_account_info(),
+            token::Transfer {
+                from: self.vault.to_account_info(),
+                to: self.withdraw_destination.to_account_info(),
+                authority: self.authority.to_account_info(),
+            },
+        )
```
[the remark this answers is not in the document]
Comment (0xCipherCoder) [reply content/courses/program-security/pda-sharing.md]: Removed impl and created separate methods get_transfer_ctx and get_secure_transfer_ctx. Fixed and pushed this changes in unbox PRs.

File: content/courses/program-security/pda-sharing.md (hunk 1 above)
Comment (0xCipherCoder) [reply content/courses/program-security/pda-sharing.md]: Imported required functions only and updated Unbox PRs as well for the same.

**Comment by 0xCipherCoder** [comment]
> Great work, some minor changes but not many, we should be able to get this in immediately once done.

Thanks! @mikemaccana I have updated code as per comment and respective Unbox PRs in both starter and solution branch.

**Comment by mikemaccana** [comment]
Thanks @0xCipherCoder . You've won this one too.

**Comment by 0xCipherCoder** [comment]
Thanks! @mikemaccana

**Pull request merged**